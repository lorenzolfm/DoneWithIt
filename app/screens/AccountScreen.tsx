import React from 'react';
import { ListItem } from '../components/ListItem';

import { Screen } from '../components/Screen';

export const AccountScreen = () => {
  return (
    <Screen>
      <ListItem
        title="Mosh Hamedani"
        subtitle="programmingwithmosh@gmail.com"
      />
    </Screen>
  );
};
