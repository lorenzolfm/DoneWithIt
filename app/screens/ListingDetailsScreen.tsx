import React from 'react';
import { Image, View, StyleSheet } from "react-native";

import { ListItem } from '../components/ListItem';
import { AppText } from "../components/AppText";

import defaultStyles from '../config/styles'

export const ListingDetailsScreen = ({ route }) => {
  const listing = route.params;

  return (
    <View>
      <Image style={styles.image} source={listing.image} />
      <View style={styles.detailsContainer}>
        <AppText style={styles.title}>{listing.title}</AppText>
        <AppText style={styles.price}>{listing.price}</AppText>
        <View style={styles.userContainer}>
          <ListItem
            image={require('../assets/mosh.jpg')}
            title="Mosh Hamedani"
            subtitle="5 Listings"
          />
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  image: {
    width: '100%',
    height: 300,
  },
  detailsContainer: {
    padding: 20,
  },
  title: {
    fontSize: 24,
    fontWeight: '500',
  },
  price: {
    color: defaultStyles.colors.secondary,
    fontWeight: 'bold',
    fontSize: 20,
    marginVertical: 10,
  },
  userContainer: {
    marginVertical: 40,
  }
});
