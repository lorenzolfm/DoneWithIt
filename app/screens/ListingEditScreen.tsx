import React, { useState } from 'react';
import { StyleSheet } from 'react-native';
import * as Yup from 'yup';
import { useLocation } from '../hooks/useLocation';
import { addListing } from '../api/listings'

import {
  AppForm,
  AppFormField,
  AppFormPicker,
  SubmitButton
} from '../components/Forms'
import { CategoryPickerItem } from '../components/CategoryPickerItem';
import { Screen } from '../components/Screen';
import { FormImagePicker } from '../components/Forms/FormImagePicker';
import { Listing } from '../types';
import { UploadScreen } from './UploadScreen';

const validationSchema = Yup.object().shape({
  title: Yup.string().required().min(1).label('Title'),
  price: Yup.number().required().min(1).max(10000).label('Price'),
  description: Yup.string().label('description'),
  category: Yup.object().required().nullable().label('Category'),
  images: Yup.array().min(1, "Please select at least one image"),
});

const categories = [
  { label: 'Furniture', value: 1, backgroundColor: 'red', icon: 'apps' },
  { label: 'Clothing', value: 2, backgroundColor: 'green', icon: 'email'},
  { label: 'Camera', value: 3, backgroundColor: 'blue', icon: 'lock' },
];

const initialValues = {
  title: '',
  price: '',
  description: '',
  category: null,
  images: [],
};

export const ListingEditScreen = () => {
  const location = useLocation();
  const [uploadVisible, setUploadVisible] = useState<boolean>(false);
  const [progress, setProgress] = useState<number>(0);

  const handleSubmit = async (listing: Listing) => {
    setProgress(0)
    setUploadVisible(true);
    const result = await addListing(
      {...listing, location},
      progress => setProgress(progress)
    );
    setUploadVisible(false);

    if (!result.ok) {
      setUploadVisible(false);
      console.log(result.problem)
      return alert("Could not save the listing");
    }

    alert('Success');
  }

  return (
    <Screen style={styles.container}>
      <UploadScreen progress={progress} visible={uploadVisible} />
      <AppForm
        initialValues={initialValues}
        onSubmit={handleSubmit}
        validationSchema={validationSchema}
      >
        <FormImagePicker name="images" />
        <AppFormField maxLength={255} name="title" placeholder="Title" />
        <AppFormField
          keyboardType="numeric"
          maxLength={8}
          name="price"
          placeholder="Price"
          width={120}
        />
        <AppFormPicker
          items={categories}
          numberOfColumns={3}
          name="category"
          PickerItemComponent={CategoryPickerItem}
          placeholder="Category"
          width="50%"
        />
        <AppFormField
          maxLength={255}
          multiline
          name="description"
          numberOfLines={3}
          placeholder="Description"
        />
        <SubmitButton title="Post"/>
      </AppForm>
    </Screen>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '95%',
    alignSelf: 'center',
    marginTop: 10,
  },
})
