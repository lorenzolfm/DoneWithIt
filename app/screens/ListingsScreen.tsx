import React, { useEffect } from 'react';
import { FlatList, StyleSheet } from "react-native";

import { Card } from '../components/Card';
import { Screen } from "../components/Screen";
import { ActivityIndicator } from '../components/ActivityIndicator';
import { Navigation } from '../types'
import routes from '../navigation/routes';
import { getListings } from '../api/listings';

import defaultStyles from '../config/styles';
import { AppText } from '../components/AppText';
import { Button } from '../components/Button';
import { useApi } from '../hooks/useApi';

export const ListingsScreen = ({ navigation: { navigate } }: Navigation) => {
  const { data: listings, error, loading, request: loadListings } = useApi(getListings)

  useEffect(() => {
    loadListings();
  }, [])

  return (
    <Screen style={styles.screen}>
      {error && <>
        <AppText>Couldn't retrieve listings</AppText>
        <Button onPress={loadListings} title="Retry" />
      </>}
      <ActivityIndicator visible={loading}/>
      <FlatList
        data={listings}
        keyExtractor={listing => listing.id.toString()}
        renderItem={({ item }) =>
          <Card
            title={item.title}
            subtitle={"$" + item.price}
            imageUrl={item.images[0].url}
            thumbnailUrl={item.images[0].thumbnailUrl}
            onPress={() => navigate(routes.LISTING_DETAILS, item)}
          />
        }
      />
    </Screen>
  );
};

const styles = StyleSheet.create({
  screen: {
    padding: 20,
    backgroundColor: defaultStyles.colors.light,
  },
});
