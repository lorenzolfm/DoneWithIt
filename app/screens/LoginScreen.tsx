import React from 'react';
import { Image, StyleSheet } from 'react-native';
import { Formik } from 'formik';
import * as Yup from 'yup';

import { AppTextInput } from '../components/AppTextInput';
import { Button } from '../components/Button';
import { Screen } from '../components/Screen';
import { AppText } from '../components/AppText';

const validationSchema = Yup.object().shape({
  email: Yup.string().required().email().label('Email'),
  password: Yup.string().required().min(4).label('Password'),
});

const initialValues = {
  email: '',
  password: '',
};

export const LoginScreen = () => {
  return (
    <Screen style={styles.container}>
      <Image source={require("../assets/logo-red.png")} style={styles.logo}/>
      <Formik
        initialValues={initialValues}
        onSubmit={values => console.log(values)}
        validationSchema={validationSchema}
      >
        { ({ handleChange, handleSubmit, errors }) => (
          <>
            <AppTextInput
              autoCapitalize="none"
              autoCorrect={false}
              keyboardType="email-address"
              onChangeText={handleChange('email')}
              icon="email"
              placeholder="Email"
              textContentType="email"
            />
            <AppText style={{ color: 'red' }}>{errors.email}</AppText>
            <AppTextInput
              autoCapitalize="none"
              autoCorrect={false}
              icon="lock"
              onChangeText={handleChange('password')}
              placeholder="Password"
              textContentType="password"
              secureTextEntry
            />
            <AppText style={{ color: 'red' }}>{errors.password}</AppText>
            <Button title="Login" onPress={() => handleSubmit()} />
          </>
        )}
      </Formik>
    </Screen>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 10,
  },
  logo: {
    width: 80,
    height: 80,
    alignSelf: 'center',
    marginTop: 50,
    marginBottom: 20,
  },
});
