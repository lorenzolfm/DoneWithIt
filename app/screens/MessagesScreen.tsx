import React, { useState } from 'react';
import { FlatList } from "react-native"

import { ListItem } from '../components/ListItem';
import { ListItemDeleteAction } from '../components/ListItemDeleteAction';
import { ListItemSeparator } from '../components/ListItemSeparator';
import { Screen } from '../components/Screen';

interface Message {
  id: number,
  title: string,
  description: string,
  image: any,
};

const msgs: Array<Message> = [
  {
    id: 0,
    title: 'T1',
    description: 'D1',
    image: require('../assets/mosh.jpg'),
  },
  {
    id: 1,
    title: 'T2',
    description: 'D2',
    image: require('../assets/mosh.jpg'),
  },
];

export const MessagesScreen = () => {
  const [messages, setMessages] = useState(msgs);
  const [refreshing, setRefreshing] = useState(false);

  const handleDelete = (message: Message): void => {
    setMessages(messages.filter(m => m.id !== message.id));
  };

  return (
    <Screen>
      <FlatList
        data={messages}
        keyExtractor={item => item.id.toString()}
        renderItem={
          ({ item }) =>
            <ListItem
              title={item.title}
              subtitle={item.description}
              image={item.image}
              onPress={() => console.log('Message selected', item)}
              renderRightActions={() => <ListItemDeleteAction onPress={() => handleDelete(item)} />}
            />
        }
        ItemSeparatorComponent={ListItemSeparator}
        refreshing={refreshing}
        onRefresh={() => setMessages([...msgs])}
      />
    </Screen>
  );
};
