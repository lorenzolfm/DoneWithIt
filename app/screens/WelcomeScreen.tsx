import React from 'react';
import { View, ImageBackground, StyleSheet, Image, Text } from "react-native";
import { Button } from '../components/Button';
import routes from '../navigation/routes';

export const WelcomeScreen = ({ navigation: { navigate } }) => {
  return (
    <ImageBackground
      blurRadius={10}
      source={require('../assets/background.jpg')}
      style={styles.background}
    >
      <View style={styles.logoContainer}>
        <Image source={require('../assets/logo-red.png')} style={styles.logo}/>
        <Text style={styles.tagline}>Sell What You Don't Need</Text>
      </View>
      <View style={styles.buttonsContainer}>
        <Button title="Login" onPress={() => navigate(routes.LOGIN)} />
        <Button title="Register" onPress={() => navigate('Register')} color="secondary"/>
      </View>
    </ImageBackground>
  );
};

const styles = StyleSheet.create({
  background: {
    flex: 1,
    justifyContent: 'flex-end',
  },
  logo: {
    width: 100,
    height: 100,
  },
  logoContainer: {
    alignSelf: 'center',
    alignItems: 'center',
    position: 'absolute',
    top: 70,
  },
  buttonsContainer: {
    width: '100%',
    padding: 20,
  },
  tagline: {
    fontSize: 25,
    fontWeight: '600',
    paddingVertical: 20,
  }
});
