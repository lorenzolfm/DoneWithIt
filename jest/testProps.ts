export const mockNav = jest.fn();
export const mockGoBack = jest.fn();
export const createTestProps = (props: any) => ({
  navigation: {
    navigate: mockNav,
    goBack: mockGoBack,
  },
  ...props,
});

export const props = createTestProps({});
