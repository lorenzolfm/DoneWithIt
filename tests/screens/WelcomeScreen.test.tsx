import { shallow } from 'enzyme';
import { WelcomeScreen } from '../../app/screens/WelcomeScreen';
import { props } from '../../jest/testProps';

describe('WelcomeScreen', () => {
  it('matches snapshot', () => {
    expect(shallow(<WelcomeScreen {...props} />)).toMatchSnapshot();
  });
});
