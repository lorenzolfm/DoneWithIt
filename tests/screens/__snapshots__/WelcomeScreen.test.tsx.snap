// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen matches snapshot 1`] = `
<ImageBackground
  source={
    Object {
      "testUri": "../../../app/assets/background.jpg",
    }
  }
  style={
    Object {
      "flex": 1,
      "justifyContent": "flex-end",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "position": "absolute",
        "top": 70,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../app/assets/logo-red.png",
        }
      }
      style={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    />
    <Text>
      Sell What You Don't Need
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fc5c65",
        "height": 70,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#4ecdc4",
        "height": 70,
        "width": "100%",
      }
    }
  />
</ImageBackground>
`;
